### PR TITLE
fix(genesis): separate genesis param and chain param

### LINKED ISSUE #1341: Set fee paramaetrs to zero in genesis doc
Once creating a new genesis file for the LocalNet the deprecated fee parameters set to default values, like:
```
      "fee_fraction": 0.0001,
      "minimum_fee": 1000,
      "maximum_fee": 1000000, 
```

These parameters should set to zero and mark as deprecated.

### PR DESCRIPTION
**Pull request by kehiy**
## Related issue
- Fixes #1341

### CODE REVIEW

File: types/param/genesis_param.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package param
+
+import (
+	"time"
+
+	"github.com/pactus-project/pactus/types/amount"
+)
+
+type GenParams struct {
```
Comment (b00f) [inline types/param/genesis_param.go:9]: Move this to `genesis` package and make it private

File: types/param/genesis_param.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	BlockVersion              uint8         `cbor:"1,keyasint"  json:"block_version"`
+	BlockIntervalInSecond     int           `cbor:"2,keyasint"  json:"block_interval_in_second"`
+	CommitteeSize             int           `cbor:"3,keyasint"  json:"committee_size"`
+	BlockReward               amount.Amount `cbor:"4,keyasint"  json:"block_reward"`
+	TransactionToLiveInterval uint32        `cbor:"5,keyasint"  json:"transaction_to_live_interval"`
+	BondInterval              uint32        `cbor:"6,keyasint"  json:"bond_interval"`
+	UnbondInterval            uint32        `cbor:"7,keyasint"  json:"unbond_interval"`
+	SortitionInterval         uint32        `cbor:"8,keyasint"  json:"sortition_interval"`
+	FeeFraction               float64       `cbor:"9,keyasint"  json:"fee_fraction"` // Deprecated: Replaced by fix fee
```
Comment (b00f) [inline types/param/genesis_param.go:18]:
```suggestion
	FeeFraction               float64       `cbor:"9,keyasint"  json:"_"` // Deprecated: Replaced by fix fee
```

**Review by b00f — CHANGES_REQUESTED** [review]

File: types/param/genesis_param.go (hunk 2 above)
Comment (kehiy) [reply types/param/genesis_param.go]: Ignoring deprecated fields results in changing the genesis hash. should I do that?

File: genesis/genesis_param.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+type genParams struct {
+	BlockVersion              uint8         `cbor:"1,keyasint"  json:"block_version"`
+	BlockIntervalInSecond     int           `cbor:"2,keyasint"  json:"block_interval_in_second"`
+	CommitteeSize             int           `cbor:"3,keyasint"  json:"committee_size"`
+	BlockReward               amount.Amount `cbor:"4,keyasint"  json:"block_reward"`
+	TransactionToLiveInterval uint32        `cbor:"5,keyasint"  json:"transaction_to_live_interval"`
+	BondInterval              uint32        `cbor:"6,keyasint"  json:"bond_interval"`
+	UnbondInterval            uint32        `cbor:"7,keyasint"  json:"unbond_interval"`
+	SortitionInterval         uint32        `cbor:"8,keyasint"  json:"sortition_interval"`
+	FeeFraction               float64       `cbor:"9,keyasint"  json:"fee_fraction"` // Deprecated: Replaced by fix fee
```
Comment (b00f) [inline genesis/genesis_param.go:18]:
```suggestion
	feeFraction               float64       `cbor:"9,keyasint"  json:"fee_fraction"` // Deprecated: Replaced by fix fee
```

File: genesis/genesis_param.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	MinimumFee                amount.Amount `cbor:"10,keyasint" json:"minimum_fee"`  // Deprecated: Replaced by fix fee
```
Comment (b00f) [inline genesis/genesis_param.go:19]:
```suggestion
	minimumFee                amount.Amount `cbor:"10,keyasint" json:"minimum_fee"`  // Deprecated: Replaced by fix fee
```

File: genesis/genesis_param.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	MaximumFee                amount.Amount `cbor:"11,keyasint" json:"maximum_fee"`  // Deprecated: Replaced by fix fee
```
Comment (b00f) [inline genesis/genesis_param.go:20]:
```suggestion
	maximumFee                amount.Amount `cbor:"11,keyasint" json:"maximum_fee"`  // Deprecated: Replaced by fix fee
```

File: types/param/param.go [hunk 6]
```diff
@@ -15,11 +15,9 @@ type Params struct {
 	BondInterval              uint32        `cbor:"6,keyasint"  json:"bond_interval"`
 	UnbondInterval            uint32        `cbor:"7,keyasint"  json:"unbond_interval"`
 	SortitionInterval         uint32        `cbor:"8,keyasint"  json:"sortition_interval"`
-	FeeFraction               float64       `cbor:"9,keyasint"  json:"fee_fraction"` // Deprecated: Replaced by fix fee
-	MinimumFee                amount.Amount `cbor:"10,keyasint" json:"minimum_fee"`  // Deprecated: Replaced by fix fee
-	MaximumFee                amount.Amount `cbor:"11,keyasint" json:"maximum_fee"`  // Deprecated: Replaced by fix fee
-	MinimumStake              amount.Amount `cobr:"12,keyasint" json:"minimum_stake"`
-	MaximumStake              amount.Amount `cbor:"13,keyasint" json:"maximum_stake"`
+	MaxTransactionsPerBlock   int           `cbor:"9,keyasint"  json:"max_transaction_per_block"`
```
Comment (b00f) [inline types/param/param.go:18]: Remove both JSON and CBOR tags here

**Review by b00f — CHANGES_REQUESTED** [review]

**Comment by kehiy** [comment]
@b00f requested changes are done. but I can't make fee parameters private in genesis params since we will get linter errors.

**Review by b00f — APPROVED** [review]

File: node/node.go [hunk 7]
```diff
@@ -41,7 +42,7 @@ type Node struct {
 	nanomsg    *nanomsg.Server
 }
 
-func NewNode(genDoc *genesis.Genesis, conf *config.Config,
+func NewNode(genDoc *genesis.Genesis, chainParam *param.Params, conf *config.Config,
```
Comment (b00f) [inline node/node.go:45]: Why did you pass params when you can get it from genesis?

**Pull request closed without merging**